src/main/webapp/js/components/interface/drawer/TabbedDrawer.js

```diff
@@ -79,90 +78,117 @@ define(function (require) {
                 );
             }, this);
             return renderedLabels;
-        }
+        };
 
         // function to render all the childs, wrap each one of them in a div and manage with display
         // which child has to be visible.
-        renderMyChildren() {
-            var renderedComponents = this.props.children.map(function(child, index) {
-                if(this.state.drawerOpened == true) {
-                    var ElementToRender = child;
-                    var MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender /></div> : <div key={index}><ElementToRender /></div>;
-
-                } else {
-                    var ElementToRender = child;
-                    var MyElement = <div key={index}><ElementToRender className="hiddenComponent" /></div>;
-                }
-                return (MyElement);
-            }, this);
-            return renderedComponents;
-        }
+        renderMyChildren = () => {
```

Comment:
This function is a bit of black magic to me. Maybe some comments can help...

Reply:
adding some comments to explain it.

Reply:
For completeness, I tested this on netpyne using old and new method in the component Initialization. As well I gave it a try with the default extension to be sure it worked there as well, since some of the casper tests were failing.

Reply:
Thanks, very clear comments